### ISSUE TITLE
Pass `--proxy` and `--https-proxy` options to `npm`

### ISSUE THREAD

**Issue by danfuzz** [arei/npmbox#80]
By default `npmunbox` allows network requests to happen, which to be clear is legit in some circumstances. However, in many use cases the explicit desire is to fail rather than ever try to use networking. Given scripts (see issue #79), it's not going to be practical to cut off all the possibilities, but by providing a non-operational endpoint as the value for the `--proxy` and `--https-proxy` options to `npm`, it will at least be possible to keep the underlying `npm install` command from _directly_ using networking.

I suggest the option name `--kill-network` to enable this.

**Comment by arei** [comment]
I don't think the proxy info is passed to the underlying scripts.  This switch implies that it does and that npmbox has any say in what, node-gyp, for example, does... I'm happy to be wrong if you find out otherwise though.

**Comment by danfuzz** [comment]
Sorry, I meant by my description that this _would not_ guarantee that scripts would abide by it. That said, in testing it does look like it does (somehow) get respected by `node-gyp` specifically. (I'm testing with a box of `fsevents`.)

I guess the question comes down to: Is it okay for `npmunbox` to implement what amounts to a "best effort but no hard guarantee" in this regard, or would you prefer it not to implement this at all?

As an alternative to this, it would perhaps be reasonable just to allow `--proxy` and `--https-proxy` options to be passed through to the underlying `npm`, which would perhaps satisfy other reasonable use cases.

**Comment by danfuzz** [comment]
BTW, this is my current code for the option:

```javascript
if (argv["kill-network"]) {
	// Note: Port 37 is associated with the traditional "time" protocol. (a) It
	// is commonly disabled. (b) If enabled, it will not respond like an HTTP(S)
	// proxy. (c) If enabled, it also won't actually do anything dire if we send
	// garbage data to it.
	options.proxy = options["https-proxy"] = "http://127.0.0.1:37/network-disabled";
}
```

**Comment by danfuzz** [comment]
Also, ICYC, in my current context, many of the machines I'll be unboxing on _don't_ have networking disabled but _do_ trigger alerts when networking is used.

**Comment by arei** [comment]
I think I kind of like just exposing `--proxy` and `--https-proxy`...

**Comment by danfuzz** [comment]
Okeedoke!

**Comment by danfuzz** [comment]
Fixed in v4.2.0, just published.
